File: HorseRacing.Contracts/Models/Authentication/Responses/AuthenticationResponse.cs
using HorseRacing.Contracts.Base.Responses;
using HorseRacing.Contracts.Models.Authentication.Dtos;

namespace HorseRacing.Contracts.Models.Authentication.Responses
{
    /// <summary>
    /// Модель ответа при авторизации
    /// </summary>
    public class AuthenticationResponse : BaseResponse<AuthenticationResponseDto>
    {
        public AuthenticationResponse() : base() { }

        public AuthenticationResponse(AuthenticationResponseDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/auth/responses/authentication-response.ts
import type { BaseResponse } from "../../base/responses/base-response";
import type { AuthenticationResponseDto } from "../auth/dto/authentication-response-dto";

export class AuthenticationResponse implements BaseResponse<AuthenticationResponseDto> {
    Data: AuthenticationResponseDto;

    constructor(data?: AuthenticationResponseDto) {
        this.Data = data || { Id: '', Username: '', Token: '', FirstName: '', LastName: '', Email: '' };
    }
}